src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -3463,9 +3461,10 @@ void EmitSingleCharAtomicLoop(RegexNode node)
                     // .* was used with RegexOptions.Singleline, which means it'll consume everything.  Just jump to the end.
                     // The unbounded constraint is the same as in the Notone case above, done purely for simplicity.
 
-                    // int i = end - pos;
+                    // int i = inputSpan.Length - pos;
                     TransferSliceStaticPosToPos();
-                    Ldloc(end);
+                    Ldloca(inputSpan);
+                    Call(s_spanGetLengthMethod);
```

Comment:
NIT: Since we do this a lot now, should we consider a Macro that does the inputSpan.Length similar to how we have one for SliceInputSpan?

Reply:
There are already a bunch of places where we get a span's length.  If you think it's worth adding a helper for, we can go back through after and replace each pair of two calls with one call.